test/e2e/e2e_leaderelection_test.go

```diff
@@ -0,0 +1,201 @@
+	if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace1, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unable to create ns %v", testNamespace1.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace1.Name, metav1.DeleteOptions{})
+
+	t.Logf("Creating testing namespace %v", ns2)
+	testNamespace2 := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns2}}
+	if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace2, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unable to create ns %v", testNamespace2.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace2.Name, metav1.DeleteOptions{})
+
+	deployment1, err := createDeployment(ctx, clientSet, ns1, 5, t)
+	if err != nil {
+		t.Fatalf("create deployment 1: %v", err)
+	}
+	defer clientSet.AppsV1().Deployments(deployment1.Namespace).Delete(ctx, deployment1.Name, metav1.DeleteOptions{})
+
+	deployment2, err := createDeployment(ctx, clientSet, ns2, 5, t)
+	if err != nil {
+		t.Fatalf("create deployment 2: %v", err)
+	}
+	defer clientSet.AppsV1().Deployments(deployment2.Namespace).Delete(ctx, deployment2.Name, metav1.DeleteOptions{})
+
+	waitForPodsRunning(ctx, t, clientSet, map[string]string{"test": "leaderelection", "name": "test-leaderelection"}, 5, ns1)
+
+	podListAOrg := getPodNameList(ctx, clientSet, ns1, t)
+
+	waitForPodsRunning(ctx, t, clientSet, map[string]string{"test": "leaderelection", "name": "test-leaderelection"}, 5, ns2)
+
+	podListBOrg := getPodNameList(ctx, clientSet, ns2, t)
+
+	s1, err := options.NewDeschedulerServer()
+	if err != nil {
+		t.Fatalf("unable to initialize server: %v", err)
+	}
+	s1.Client = clientSet
+	s1.DeschedulingInterval = 5 * time.Second
+	s1.LeaderElection.LeaderElect = true
+	s1.KubeconfigFile = os.Getenv("KUBECONFIG")
+	s1.PolicyConfigFile = "./policy_leaderelection_a.yaml"
+
+	s2, err := options.NewDeschedulerServer()
+	if err != nil {
+		t.Fatalf("unable to initialize server: %v", err)
+	}
+	s2.Client = clientSet
+	s2.DeschedulingInterval = 5 * time.Second
+	s2.LeaderElection.LeaderElect = true
+	s2.KubeconfigFile = os.Getenv("KUBECONFIG")
+	s2.PolicyConfigFile = "./policy_leaderelection_b.yaml"
+
+	t.Log("starting deschedulers")
+
+	go func() {
+		err := descheduler.Run(ctx, s1)
+		if err != nil {
+			t.Errorf("unable to start descheduler: %v", err)
+			return
+		}
+	}()
+
+	time.Sleep(1 * time.Second)
+
+	go func() {
+		err := descheduler.Run(ctx, s2)
+		if err != nil {
+			t.Errorf("unable to start descheduler: %v", err)
+			return
+		}
+	}()
+
+	defer clientSet.CoordinationV1().Leases(s1.LeaderElection.ResourceNamespace).Delete(ctx, s1.LeaderElection.ResourceName, metav1.DeleteOptions{})
+	defer clientSet.CoordinationV1().Leases(s2.LeaderElection.ResourceNamespace).Delete(ctx, s2.LeaderElection.ResourceName, metav1.DeleteOptions{})
+
+	// wait for a while so all the pods are 5 seconds older
+	time.Sleep(5 * time.Second)
```

Comment:
7 is better since 5s is exactly the boundary given the second descheduler instance is created after `time.Sleep(1 * time.Second)`.